src/tapis-hooks/utils/infiniteQuery.ts

```diff
@@ -0,0 +1,16 @@
+import { TapisResponse, TapisPaginatedRequest } from 'tapis-api/types/TapisInterfaces';
+
+export type ResultPages<T> = { result?: T[] }[];
+
+export const concatResults = <T>(pages: ResultPages<T>): T[] =>
+  pages.reduce(
+    (accumulator: T[], current) => accumulator.concat(current.result ?? []),
+    []
+  );
+
+export const tapisNextPageParam = (
+  lastPage: TapisResponse, allPages: TapisResponse[], 
```

Comment:
```suggestion
export const tapisNextPageParam<T> = (
  lastPage: T, allPages: T[], 
```

If we pass `Files.FileListingResponse` or whatever Tapis-Typescript response type we need, we can avoid having to define a separate interface.